started on updating the markdown export

Mpdreamz opened the pull request.

Mpdreamz committed:
started on updating the markdown export

russcam commented:
If this now takes a dependency on a fork of JustAssembly, we should consider making members of ` assemblyDiffPair.Diff` public so that diff text from the object graph directly rather than from XML, or alternatively, update the diff items to allow different types of exporter, XML, Markdown, Asccidoc, etc.

Mpdreamz committed:
update justassembly

Mpdreamz committed:
update justdecompile

Mpdreamz committed:
allow the build to fail if the result exceeds needed version change

Mpdreamz committed:
update references in project

Mpdreamz commented:
@russcam I've updated the dlls and made these types public 👍 

This PR took a different focus and you can now pass a flag to prevent certain version changes e.g:

>dotnet run --framework netcoreapp3.1 -- "previous-nuget|assembly-differ|0.9.1|netcoreapp3.1" "directory|bin/Debug/netcoreapp3.1" --target=assembly-differ -f=xml -p=minor

Will fail if you introduce changes locally that exceed `patch|minor|major`. 

Will create a new PR with a new exporter specifically to be used with Github Action commenting on commits as per #4 . 

To this I introduced a new type of `IExporter` and a new type to hold the collective results. You can now write an exporter that writes a file for each `AssemblyComparison` (all the previous ones). Or one for the whole report.
